packages/cozy-realtime/src/index.spec.js

```diff
@@ -0,0 +1,239 @@
+
+    beforeEach(async () => {
+      handlerCreate = jest.fn()
+      handlerUpdate = jest.fn()
+      handlerDelete = jest.fn()
+      await realtime.onCreate({ type }, handlerCreate)
+      await realtime.onUpdate({ type }, handlerUpdate)
+      await realtime.onDelete({ type }, handlerDelete)
+      await realtime.onCreate({ type: 'io.cozy.accounts' }, handlerCreate)
+    })
+
+    afterEach(() => {
+      realtime.unsubscribeAll()
+    })
+
+    it('should unsubscribe handlerCreate with type, eventName and handler', () => {
+      expect(realtime._socket.isOpen()).toBe(true)
+      expect(realtime._numberOfHandlers).toBe(4)
+      realtime.unsubscribe({ type, eventName: EVENT_CREATED }, handlerCreate)
+      expect(realtime._numberOfHandlers).toBe(3)
+      expect(realtime._socket.isOpen()).toBe(true)
+    })
+
+    it('should unsubscribe handlerCreate with type and eventName', () => {
+      expect(realtime._socket.isOpen()).toBe(true)
+      expect(realtime._numberOfHandlers).toBe(4)
+      realtime.unsubscribe({ type, eventName: EVENT_CREATED })
+      expect(realtime._numberOfHandlers).toBe(3)
+      expect(realtime._socket.isOpen()).toBe(true)
+    })
+
+    it('should unsubscribe handlerCreate with type', () => {
+      expect(realtime._socket.isOpen()).toBe(true)
+      expect(realtime._numberOfHandlers).toBe(4)
+      realtime.unsubscribe({ type })
+      expect(realtime._numberOfHandlers).toBe(1)
+      expect(realtime._socket.isOpen()).toBe(true)
+    })
+
+    it('should unsubscribe all events', () => {
+      expect(realtime._socket.isOpen()).toBe(true)
+      expect(realtime._numberOfHandlers).toBe(4)
+      realtime.unsubscribeAll()
+      expect(realtime._numberOfHandlers).toBe(0)
+      expect(realtime._socket.isOpen()).toBe(false)
+    })
+  })
+
+  describe('events', () => {
+    it('should emit error when retry limit is exceeded', async done => {
+      realtime._retryLimit = 0
+
+      realtime.on('error', () => done())
+      const handler = jest.fn()
+
+      await realtime.onCreate({ type }, handler)
+      expect(realtime._socket.isOpen()).toBe(true)
+      cozyStack.simulate('error')
+    })
+  })
+
+  describe('authentication', () => {
+    it('should update socket authentication when client login', async () => {
+      realtime._socket.updateAuthentication = jest.fn()
+      cozyClient.emit('login')
+      expect(realtime._socket.updateAuthentication.mock.calls.length).toBe(1)
+    })
+
+    it('should update socket authentication when client token refreshed ', async () => {
+      realtime._socket.updateAuthentication = jest.fn()
+      cozyClient.emit('login')
+      expect(realtime._socket.updateAuthentication.mock.calls.length).toBe(1)
+    })
+  })
+})
+
+describe('generateKey', () => {
+  it('should return key from config', () => {
+    let config = { type: 'io.cozy.bank.accounts', eventName: EVENT_CREATED }
+    expect(generateKey(config)).toBe('io.cozy.bank.accounts\\created')
```

Comment:
I am not a fan of the usage of backslashes as separator. Backslashes have special meaning so \\ will get wrongly logged.

```
console.log('io.cozy.bank.accounts\\created')
io.cozy.bank.accounts\created
```

Common separators include forward slashes `/` or dots `.`

Reply:
I replace by '//' https://github.com/cozy/cozy-libs/pull/424/commits/f680c4f82451e2683bfd4fee6acb0a6c97a2d537